### PR TITLE
Adding `GlobalExclusiveDeviceAccess` 

### PR DESCRIPTION
**Pull request by frobijn**
## Description

- GlobalExclusiveDeviceAccess added for the execution of tests on real hardware. The test framework waits at most 5 seconds to get exclusive access, if that is not granted it will stop trying.

- .editorconfig added

## Motivation and Context

The test framework should get exclusive access as soon as possible, and keep that until the tests are finished. The framework also should have a timeout and not wait indefinite for a device to become available.

## How Has This Been Tested?

Using the poc (by reference) and simultaneous access by nanoff.

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [x] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:
- [x] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

## Summary by CodeRabbit

- **Chores**
	- Updated the version of the `nanoFramework.Tools.Debugger.Net` dependency to improve stability and performance.

### CODE REVIEW

**Review by josesimoes — APPROVED** [review]
LGTM!

File: source/TestAdapter/Executor.cs [hunk 1]
```diff
@@ -195,335 +196,342 @@ private async Task<List<TestResult>> RunTestOnHardwareAsync(List<TestCase> tests
...
-                {
-                    results.First().Outcome = TestOutcome.Failed;
-                    results.First().ErrorMessage = "Couldn't find any valid nanoDevice. Maybe try to disable the device watchers in Visual Studio Extension! If the situation persists reboot the device and/or disconnect and connect it again.";
+                        // all good here, proceed to execute tests
+                        goto executeTests;
+                    }
+#if DEBUG
+                    catch (Exception ex)
+#else
+                catch
```
Comment (Ellerbach) [inline source/TestAdapter/Executor.cs:236]: extremly minimal: the indention looks a bit odd for this line

File: source/TestAdapter/Executor.cs [hunk 2]
```diff
@@ -195,335 +196,342 @@ private async Task<List<TestResult>> RunTestOnHardwareAsync(List<TestCase> tests
...
-                    // only class libs have this attribute, therefore sanity check is required
-                    string nativeVersion = "";
-                    if (match.Count == 1)
+                // check if device is still in initialized state
+                if (!deviceIsInInitializeState)
+                {
+                    // device has left initialization state
+                    _logger.LogMessage($"Device is initialized and ready!", Settings.LoggingLevel.Verbose);
+                    await Task.Yield();
+
```
Comment (Ellerbach) [inline source/TestAdapter/Executor.cs:438]: very minimal: extra line here

File: source/TestAdapter/Executor.cs [hunk 3]
```diff
@@ -195,335 +196,342 @@ private async Task<List<TestResult>> RunTestOnHardwareAsync(List<TestCase> tests
...
+                        // AssemblyNativeVersion
+                        pattern = @"(?<=AssemblyNativeVersion\("")(.*)(?=\""\)])";
+                        match = Regex.Matches(assemblyProperties, pattern, RegexOptions.IgnoreCase);
 
-                _logger.LogMessage($"Deploying {assemblyList.Count:N0} assemblies to device... Total size in bytes is {totalSizeOfAssemblies}.", Settings.LoggingLevel.Verbose);
-                // need to keep a copy of the deployment blob for the second attempt (if needed)
-                var assemblyCopy = new List<byte[]>(assemblies);
+                        // only class libs have this attribute, therefore sanity check is required
+                        string nativeVersion = "";
```
Comment (Ellerbach) [inline source/TestAdapter/Executor.cs:489]:
```suggestion
                        string nativeVersion = string.Empty;
```

**Review by Ellerbach — APPROVED** [review]
Looks good, couple of minor comments.

**Pull request merged**